cmd/ocm/list/rhRegion/cmd.go

```diff
@@ -18,24 +27,51 @@ ocm list rh-regions`,
 	Hidden: true,
 }
 
+func init() {
+	flags := Cmd.Flags()
+	flags.StringVar(
+		&args.discoveryURL,
+		"discovery-url",
+		"",
+		"URL of the OCM API gateway. If not provided, will reuse the URL from the configuration "+
+			"file or "+sdk.DefaultURL+" as a last resort. The value should be a complete URL "+
+			"or a valid URL alias: "+strings.Join(urls.ValidOCMUrlAliases(), ", "),
+	)
+}
+
 func run(cmd *cobra.Command, argv []string) error {
-	// Load the configuration file:
-	cfg, err := config.Load()
-	if err != nil {
-		return fmt.Errorf("Can't load config file: %v", err)
-	}
-	if cfg == nil {
-		return fmt.Errorf("Not logged in, run the 'login' command")
+
+	cfg, _ := config.Load()
+
+	// URL Precedent (from highest priority to lowest priority):
+	//   1. runtime `--url` cli arg (key found in `urlAliases`)
+	//   2. runtime `--url` cli arg (non-empty string)
+	//   3. config file `URL` value (non-empty string)
+	//   4. sdk.DefaultURL
+	gatewayURL := sdk.DefaultURL
+	if args.discoveryURL != "" {
+		gatewayURL = args.discoveryURL
+		if _, ok := urls.OCMURLAliases[args.discoveryURL]; ok {
+			gatewayURL = urls.OCMURLAliases[args.discoveryURL]
+		}
+	} else if cfg != nil && cfg.URL != "" {
+		// re-use the URL from the config file
+		gatewayURL = cfg.URL
 	}
 
-	regions, err := sdk.GetRhRegions(cfg.URL)
+	fmt.Fprintf(os.Stdout, "Discovery URL: %s\n\n", gatewayURL)
+	regions, err := sdk.GetRhRegions(gatewayURL)
 	if err != nil {
 		return fmt.Errorf("Failed to get OCM regions: %w", err)
 	}
 
-	for regionName := range regions {
-		fmt.Println(regionName)
+	writer := tabwriter.NewWriter(os.Stdout, 0, 0, 2, ' ', tabwriter.TabIndent)
```

Comment:
Can we use dump.Pretty() here? https://github.com/openshift-online/ocm-cli/blob/main/pkg/dump/dump.go#L33

Reply:
Im not writing the raw json, which is what dump seems to be mostly for. Theres a pretty even mix of usage between dump and tabwriter in the repo. In this case, I was copying the pattern from https://github.com/openshift-online/ocm-cli/blob/main/cmd/ocm/list/region/cmd.go#L106 which seemed like the most similar command. 

A more sophisticated cli might support variable output, like `oc get ... -o json` so we could easily pick and choose which format we wanted, but that doesnt seem to be a first class concept in ocm at the moment. For now ill leave it as the `tabwriter` unless someone else comes along with a stronger opinion.